clang/lib/CIR/Dialect/Transforms/TargetLowering/Targets/NVPTX.cpp

```diff
@@ -63,6 +66,58 @@ class NVPTXTargetLoweringInfo : public TargetLoweringInfo {
 
 } // namespace
 
+bool NVPTXABIInfo::isScalarType(mlir::Type ty) const {
+  // OG checks if:
+  //     getKind() > BuiltinType::Void && getKind() <= BuiltinType::NullPtr
+  // which is confusing and unavailable in CIR.
+  // Here we manually list out all corresponding types in CIR.
+  return isa<IntType, BoolType, SingleType, DoubleType, FP16Type, FP128Type,
+             FP80Type, DataMemberType, PointerType>(ty);
+}
+
+ABIArgInfo NVPTXABIInfo::classifyReturnType(mlir::Type ty) const {
+  if (isa<VoidType>(ty))
+    return ABIArgInfo::getIgnore();
+
+  if (getContext().getLangOpts().OpenMP)
+    llvm_unreachable("NYI");
+
+  if (!isScalarType(ty))
+    return ABIArgInfo::getDirect();
+
+  // OG treats enums as their underlying type.
+  // This has already been done for CIR.
+
+  // Integers with size < 32 must be extended to 32 bits.
+  // (See Section 3.3 of PTX ABI.)
+  return (isPromotableIntegerTypeForABI(ty) ? ABIArgInfo::getExtend(ty)
+                                            : ABIArgInfo::getDirect());
+}
+
+ABIArgInfo NVPTXABIInfo::classifyArgumentType(mlir::Type ty) const {
+  if (isAggregateTypeForABI(ty)) {
```

Comment:
We don't need curly braces for single if statements, see LLVM coding style. Please fix those all around